File: ScoreCounter/app/src/main/java/com/sport/scorecounter/DownloadImageTask.java
package com.sport.scorecounter;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.drawable.BitmapDrawable;
import android.graphics.drawable.Drawable;
import android.os.AsyncTask;
import android.util.Log;
import android.widget.ImageView;
import android.widget.LinearLayout;

import java.io.InputStream;

class DownloadImageTask extends AsyncTask<String, Void, Bitmap> {

    LinearLayout bmImage;
    Drawable d;
    Bitmap mIcon11 = null;

    public DownloadImageTask(LinearLayout bmImage) {
        this.bmImage = bmImage;
    }

    protected Bitmap doInBackground(String... urls) {
        String urldisplay = urls[0];
        if (mIcon11 == null) {
            try {
                InputStream in = new java.net.URL(urldisplay).openStream();
                mIcon11 = BitmapFactory.decodeStream(in);
            } catch (Exception e) {
                Log.e("Error", e.getMessage());
                e.printStackTrace();
            }
            return mIcon11;
        }else return mIcon11;
    }

    protected void onPostExecute(Bitmap result) {
        d = null;
        Context context = this.bmImage.getContext();
        d = new BitmapDrawable(context.getResources(), result);
        System.out.println("-------------------------пришло");

        bmImage.setBackground(d);
    }
}

File: KTmvvmQuize/app/src/main/java/com/sport/ktmvvmquize/DownloadImageTask.kt
package com.sport.ktmvvmquize

import android.graphics.Bitmap
import android.graphics.drawable.BitmapDrawable
import android.os.AsyncTask
import android.widget.LinearLayout
import android.widget.RelativeLayout
import com.squareup.picasso.Picasso
import java.lang.ref.WeakReference

class DownloadImageTask(linearLayout: LinearLayout) : AsyncTask<String, Void, Bitmap?>() {

    private val linearLayoutReference: WeakReference<LinearLayout> = WeakReference(linearLayout)

    override fun doInBackground(vararg urls: String): Bitmap? {
        try {
            val imageUrl = urls[0]
            return Picasso.get().load(imageUrl).get()
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return null
    }

    override fun onPostExecute(result: Bitmap?) {
        val linearLayout = linearLayoutReference.get()
        if (linearLayout != null && result != null) {
            // Установка изображения в качестве фона для LinearLayout
            val drawable = BitmapDrawable(linearLayout.resources, result)
            linearLayout.background = drawable
        }
    }

}